### PR TITLE
[SYCL] Enabled more data types for oneMKL's gemm_batch API

### PR DESCRIPTION
**Pull request by OuadiElfarouki**
Additional `gemm_batch` types have been enabled in oneMKL (https://github.com/oneapi-src/oneMKL/pull/466) and this patch enables their corresponding APIs for the SYCL backend which eliminates the extra-steps needed when targetting NON INTEL devices to cast/copy input/output to the supported types.

The enablement of `gemm_batch_impl<sycl::half, sycl::half, float, float>` for instance removes the overhead of calling `gemm_batch_impl<sycl::half, sycl::half, sycl::half, sycl::half>` followed by a `to_fp32_sycl` for the `dst` to be copied back from `fp16` to `fp32`, which directly affects the KQ + KQV multi-batch path in quantized models Prompt Processing for instance.

Performance on intel GPUs remain the same, and a slight improvement in Prompt Processing performance on some Nvidia GPUs was observed _(0 to 3% on average)_.

- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [ ] Low
  - [x] Medium
  - [ ] High

### CODE REVIEW

File: ggml/src/ggml-sycl/dpct/helper.hpp [hunk 1]
```diff
@@ -2426,6 +2426,7 @@ namespace dpct
                                            b, ldb, beta, c, ldc, batch_size);
             break;
         }
+#endif
```
Comment (AidanBeltonS) [inline ggml/src/ggml-sycl/dpct/helper.hpp:2429]: Do you think it is worth creating an issue to inform the DPCT repo regarding the change in available dtypes?

**Review by AidanBeltonS — APPROVED** [review]

**Comment by airMeng** [comment]
https://github.com/oneapi-src/oneMKL/pull/466 is merged last week, shall you wait the the next release of oneMKL? Sorry I am not familiar with oneMKL.

File: ggml/src/ggml-sycl/dpct/helper.hpp (hunk 1 above)
Comment (OuadiElfarouki) [reply ggml/src/ggml-sycl/dpct/helper.hpp:2429]: Good idea @AidanBeltonS

**Comment by OuadiElfarouki** [comment]
@airMeng Thanks for the suggestion. At the moment there is no clear/official release process on oneMKL Interface side. We don't mention anything related to oneMKL Interface releases in the README-sycl.md as well so from a user perspective it shouldn't be confusing at the moment.
We shall adopt a different approach whenver we hear from the oneMKL side regarding their release process, so will keep this in mind!

**Comment by OuadiElfarouki** [comment]
@airMeng @joeatodd anything else we want to address for this ?

**Review by airMeng — APPROVED** [review]

**Review by joeatodd — APPROVED** [review]
LGTM :ship:

**Pull request merged**